frontend/js/src/year-in-music/2022/ComponentToImage.tsx

```diff
@@ -0,0 +1,207 @@
+import * as htmlToImage from "html-to-image";
+import React, {
+  ForwardedRef,
+  forwardRef,
+  useCallback,
+  useRef,
+  useState,
+} from "react";
+import { faCamera, faHeadphones } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { IconProp } from "@fortawesome/fontawesome-svg-core";
+import ListenCard from "../../listens/ListenCard";
+import { getEntityLink } from "../../stats/utils";
+import Loader from "../../components/Loader";
+
+export type ComponentToImageProps = {
+  data: any[];
+  entityType: Entity;
+  user: ListenBrainzUser;
+};
+
+function ComponentToImage({ data, entityType, user }: ComponentToImageProps) {
+  const [isLoading, setIsLoading] = useState(false);
+  const saveAs = (blob: string, fileName: string) => {
+    const elem = window.document.createElement("a");
+    elem.href = blob;
+    elem.download = fileName;
+    (document.body || document.documentElement).appendChild(elem);
+    if (typeof elem.click === "function") {
+      elem.click();
+    } else {
+      elem.target = "_blank";
+      elem.dispatchEvent(
+        new MouseEvent("click", {
+          view: window,
+          bubbles: true,
+          cancelable: true,
+        })
+      );
+    }
+    URL.revokeObjectURL(elem.href);
+    elem.remove();
+    setIsLoading(false);
+  };
+
+  const exportAsPicture = () => {
+    setIsLoading(true);
+    const targetId = `savable-${entityType}-component`;
+    const element = document.getElementById(targetId);
+
+    htmlToImage
+      .toPng(element as HTMLElement, {
+        canvasWidth: element!.offsetWidth,
+        canvasHeight: element!.offsetHeight,
+      })
+      .then((image) => {
+        const fileName = `${user.name}-top-${entityType}s-2021.png`;
+        if (navigator.canShare) {
+          const files = [
+            new File([image], fileName, {
+              type: "image/png",
+              lastModified: new Date().getTime(),
+            }),
+          ];
+          if (navigator.canShare({ files })) {
+            navigator
+              .share({
+                files,
+                title: `${user.name}'s Year In Music 2021`,
+                text: `Your top ${entityType}s of 2021`,
```

Comment:
```suggestion
                title: `${user.name}'s Year In Music 2022`,
                text: `Your top ${entityType}s of 2022`,
```

Reply:
Although I think I might reimplement this component and pass title and text in props, so not critical